mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -1534,15 +1534,49 @@ LogicalResult tosa::PadOp::verify() {
   if (!inputType || !outputType)
     return success();
 
-  auto paddingRank = cast<tosa::shapeType>(getPadding().getType()).getRank();
+  auto inputRank = inputType.getRank();
+  auto outputRank = outputType.getRank();
+  if (inputRank != outputRank)
+    return emitOpError() << "expect same input and output tensor rank, but got "
+                         << "inputRank: " << inputRank
+                         << ", outputRank: " << outputRank;
+
+  DenseIntElementsAttr paddingAttr;
+  if (!matchPattern(getPadding(), m_Constant(&paddingAttr)))
+    return failure();
```

Comment:
I think this should be success, since a non constant padding input is possible when CTC requirements are dropped. The validation pass should take care of checking it is indeed constant later on

Reply:
Done

Reply:
I realsed that validation pass doesn't check `padding` with `checkConstantOperandPad`.
Revert to `failure()` here.

Reply:
Ah, that'll be because `pad` is a `const_shape` which currently cannot be variable. Happy to leave as a failure in that case